File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        chars = {}
        
        for i in range(len(s)):
            chars[s[i]] = chars.get(s[i], 0) + 1
            chars[t[i]] = chars.get(t[i], 0) - 1

        for val in chars.values():
            if val != 0:
                return False

        return True

File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<int, int> seen;

        if(s.length() != t.length()){
            return false;
        }

        for(int i = 0; i < s.length(); i++){
            seen[s[i]]++;
            seen[t[i]]--;
        }

        for(const auto [key, value]: seen){
            if(value != 0)
                return false;
        }

        return true;
        
    }
};